src/sage/categories/magmatic_algebras.py

```diff
@@ -225,6 +225,61 @@ def _product_from_product_on_basis_multiply( self, left, right ):
 
         class FiniteDimensional(CategoryWithAxiom_over_base_ring):
             class ParentMethods:
+                @cached_method
+                def to_fda(self, names='e', assume_associative=True, assume_unital=True):
```

Comment:
```suggestion
                def to_finite_dimensional_algebra((self, names='e', assume_associative=True, assume_unital=True):
```
Good to be verbose with the name. I don't see a need to cache this either.

Reply:
Why not cache? This constructs a new object, and we might not want to construct it many times. Then again, a careful user should call this method once and then keep the object... What is the house style on this?

Reply:
If they don't, then they should store it themselves. This isn't something I expect someone to regularly construct, but it is quite a heavy object (in terms of memory) to keep nailed to the original algebra's existence. It also reduces the chance of a memory leak.